assets/js/modules/analytics/dashboard/dashboard-widget-analytics-adsense-top-pages.js

```diff
@@ -146,24 +144,39 @@ const isDataZero = () => {
  *
  */
 const getDataError = ( data ) => {
-	if ( ! data || ! data.error ) {
-		return false;
-	}
+	if ( data.code && data.message && data.data && data.data.status ) {
+		// Specifically looking for string "badRequest"
+		if ( data.data.reason && 'badRequest' === data.data.reason ) {
```

Comment:
This could be simplified into a single conditional:
```js
const { code, message, data, reason } = data;
if ( code && message && status && data && 'badRequest' === reason ) {
```
No need to check for `reason` and `reason ===` as it will simply be `undefined` if not there since `data` is already checked for.

Reply:
I'll remove the `data.data.reason` check since that's unnecessary, however the two clauses itself should remain separate because the first one in itself also results in `data.message` to be returned (see below).